### PR TITLE
[FIX] for local testing and fix wrong variable attribution

### PR DESCRIPTION
**Pull request by blaggacao**
**What is it**
It's a bug fix. So there was a bug beforehand.

**Why is it**
- `repo_dir = os.environ.get("TRAVIS_BUILD_DIR", ".")` this is ok. `TRAVIS_BUILD_DIR` usually coincides with '.' (seen from the `.travis.yaml`, see `coverage run --append ./travis/self_tests`)
- But already `to_preinstall = get_test_dependencies(repo_dir , addons_list)` fails as `get_test_dependencies` is not recursive, it needs the actual addons dir with is at `./tests/repo_dir`.
- And so forth ...
The only exception is
- `getaddons.get_modules_changed(repo_dir)` which works on the folder which contains the `.git` folder.

With these changes, I was able to seamlessly run the test script locall.

image

### CODE REVIEW

**Comment by blaggacao** [comment]
On the other hand, the intent to link a gitignore to another gitignore did not work in my case...

**Comment by blaggacao** [comment]
@moylop260 Could we push this true, as it would make my fully test compliant work on MQT a lot easier, faster, less error prone and less noisy?

File: .gitignore [hunk 1]
```diff
@@ -1 +1,59 @@
-sample_files/.gitignore
\ No newline at end of file
+# Byte-compiled / optimized / DLL files
```
Comment (moylop260) [inline .gitignore]: Is this change required?

Comment (blaggacao) [inline .gitignore]: Yep, this linking does not work on all (my) platform(s). Greetings!

File: .gitignore [hunk 2]
```diff
@@ -1 +1,59 @@
-sample_files/.gitignore
```
Comment (blaggacao) [inline .gitignore]: I mean this linking here...
